subprojects/core/src/main/java/org/gradle/internal/scan/config/BuildScanConfigManager.java

```diff
@@ -105,11 +108,14 @@ public BuildScanConfig collect(BuildScanPluginMetadata pluginMetadata) {
             throw new IllegalStateException("Configuration has already been collected.");
         }
 
+        VersionNumber pluginVersion = VersionNumber.parse(pluginMetadata.getVersion()).getBaseVersion();
+        if (isNotSupported(pluginVersion)) {
+            throw new UnsupportedBuildScanPluginVersionException(UNSUPPORTED_PLUGIN_VERSION_MESSAGE);
```

Comment:
IIUC, that makes every Gradle build fail if the applied build scan plugin doesn't work with the current build. I am fine with that solution, although I was expecting something to only make the `gradle/gradle` build fail if the build scan plugin is not supported.